traitlets/config/configurable.py

```diff
@@ -203,8 +210,13 @@ def _config_changed(self, change):
         section_names = self.section_names()
         self._load_config(change.new, traits=traits, section_names=section_names)
 
-    def update_config(self, config):
-        """Update config and load the new values"""
+    def update_config(self, config, skip_env=False):
```

Comment:
Maybe `override_env=True` as the default?  `skip_env` is ambigous as to whether it means "skip loading environment variables" or "skip config also set by environment variables".

Also, the default behavior should be reversed. Calling `.update_config` explicitly should load all of the config by default. The exception is loading default config *files*, which should be applied at lower-than-env priority.